### PR TITLE
Add 96Boards STM32 Sensor Mezzanine board support

### PR DESCRIPTION
**Pull request by Mani-Sadhasivam**
Add board support for 96Boards STM32 Sensor Mezzanine board. This board
is one of the 96Boards Mezzanine board which can work standalone or on
top of the 96Boards Consumer Edition boards.

Below are some of the key features of this board:

1. STM32F446VE MCU
2. 96Boards high/low-speed connectivity
3. Arduino™ Uno revision 3 and Grove connectivity
4. Integrated MEMS sensors:
   1. LSM6DS3H
   2. LIS3MDL
   3. LPS22HB
   4. MP34DT01-M
5. On-board ST-LINK/V2-1 debugger/programmer

More information about this board can be found in the 96Boards product
page: https://www.96boards.org/documentation/mezzanine/stm32/

Signed-off-by: Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, thanks.

File: boards/arm/96b_stm32_sensor_mez/96b_stm32_sensor_mez.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+identifier: 96b_stm32_sensor_mez
+name: 96B_STM32_SENSOR_MEZ
+type: mcu
+arch: arm
+toolchain:
+  - zephyr
+  - gnuarmemb
+supported:
+  - rtc
```
Comment (erwango) [inline boards/arm/96b_stm32_sensor_mez/96b_stm32_sensor_mez.yaml]: I think there are much more you can mention here (gpio, spi, ..)

File: boards/arm/96b_stm32_sensor_mez/96b_stm32_sensor_mez_defconfig [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+CONFIG_ARM_MPU=y
+
+CONFIG_SERIAL=y
+
+# console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# enable I2C
+CONFIG_I2C=y
```
Comment (erwango) [inline boards/arm/96b_stm32_sensor_mez/96b_stm32_sensor_mez_defconfig]: Disable by default

**Review by erwango — CHANGES_REQUESTED** [review]
Few changes requested, otherwise sounds good.
Do you have any plans for the missing sensors? Btw, you might be interested by https://github.com/zephyrproject-rtos/zephyr/pull/11188 (maybe already aware of).

File: boards/arm/96b_stm32_sensor_mez/96b_stm32_sensor_mez.yaml (hunk 1 above)
Comment (Mani-Sadhasivam) [reply boards/arm/96b_stm32_sensor_mez/96b_stm32_sensor_mez.yaml]: Ack

File: boards/arm/96b_stm32_sensor_mez/96b_stm32_sensor_mez_defconfig (hunk 2 above)
Comment (Mani-Sadhasivam) [reply boards/arm/96b_stm32_sensor_mez/96b_stm32_sensor_mez_defconfig]: Ack

**Comment by Mani-Sadhasivam** [comment]
@erwango I don't have any concrete plans to add the missing on-board sensors. But I may get to it at some point of time. Btw, the PR you mentioned looks interesting! I'm not aware of that. And it'd be good if I get CCed for all 96Boards related stuff.

**Comment by erwango** [comment]
@Mani-Sadhasivam, Sure! maybe we can create a 96Boards tag?

**Review by erwango — APPROVED** [review]

**Comment by Mani-Sadhasivam** [comment]
> @Mani-Sadhasivam, Sure! maybe we can create a 96Boards tag?

@erwango Yes, that makes sense!

**Review by galak — APPROVED** [review]

**Pull request merged**